### PR TITLE
[NFC][Build System: build-script] Re-organize build-script into more logical sections.

### PR DESCRIPTION
**Pull request by Rostepher**

### CODE REVIEW

**Comment by Rostepher** [comment]
@swift-ci please test

File: utils/build-script [hunk 1]
```diff
@@ -1,15 +1,20 @@
...
+# Copyright (c) 2014 - 2020 Apple Inc. and the Swift project authors
 # Licensed under Apache License v2.0 with Runtime Library Exception
 #
 # See https://swift.org/LICENSE.txt for license information
 # See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
 
-from __future__ import print_function
+
+"""
+The ultimate tool for building Swift.
```
Comment (compnerd) [inline utils/build-script]: lol, I respectfully disagree, the ultimate tool is CMake!

File: utils/build-script [hunk 2]
```diff
@@ -26,30 +31,47 @@ from build_swift.build_swift import presets
...
+from swift_build_support.swift_build_support import shell
+from swift_build_support.swift_build_support import targets
+from swift_build_support.swift_build_support import workspace
 from swift_build_support.swift_build_support.SwiftBuildSupport import (
     HOME,
     SWIFT_BUILD_ROOT,
     SWIFT_REPO_NAME,
-    SWIFT_SOURCE_ROOT,
-)
+    SWIFT_SOURCE_ROOT)
```
Comment (compnerd) [inline utils/build-script]: I think that this hunk should be dropped, the trailing comma is useful

File: utils/build-script [hunk 3]
```diff
@@ -26,30 +31,47 @@ from build_swift.build_swift import presets
...
 from swift_build_support.swift_build_support.cmake import CMake
 from swift_build_support.swift_build_support.host_specific_configuration \
     import HostSpecificConfiguration
-from swift_build_support.swift_build_support.targets import \
-    StdlibDeploymentTarget
+from swift_build_support.swift_build_support.targets import (
+    StdlibDeploymentTarget)
```
Comment (compnerd) [inline utils/build-script]: Can we drop the parenthesis?

File: utils/build-script [hunk 4]
```diff
@@ -26,30 +31,47 @@ from build_swift.build_swift import presets
...
+# -----------------------------------------------------------------------------
+# Helpers
+
+def clean_delay():
+    """Provide a short delay so accidentally invoked clean builds can be
+    canceled.
+    """
+
+    sys.stdout.write('Starting clean build in  ')
+    for i in range(3, 0, -1):
```
Comment (compnerd) [inline utils/build-script]: Im not sure I find there to be value in the `range` over `(3, 2, 1)`

File: utils/build-script (hunk 1 above)
Comment (Rostepher) [reply utils/build-script]: I was just transferring the comment from the top of the file.

File: utils/build-script (hunk 2 above)
Comment (Rostepher) [reply utils/build-script]: I intend to move all these variables to `build_swift` in the _very_ near future, trailing comma or not.

File: utils/build-script (hunk 3 above)
Comment (Rostepher) [reply utils/build-script]: I put the parens there to avoid the ugly `\` escape.

File: utils/build-script (hunk 4 above)
Comment (Rostepher) [reply utils/build-script]: I merely moved this chunk of code from the bottom of the file to the top section. I would rather not change its functionality in this PR.

**Comment by Rostepher** [comment]
@swift-ci please test

File: utils/build-script (hunk 2 above)
Comment (compnerd) [reply utils/build-script]: Sure, but, I think that preserving the comma as it moves is still valuable.

File: utils/build-script (hunk 3 above)
Comment (compnerd) [reply utils/build-script]: Ah, I see.  What about adding a trailing comma and moving the closing parenthesis to the next line?

File: utils/build-script (hunk 4 above)
Comment (compnerd) [reply utils/build-script]: Sounds reasonable.

**Pull request merged**